python/cudf/cudf/core/series.py

```diff
@@ -3009,9 +3008,11 @@ def astype(self, dtype, copy=False, errors="raise"):
             return self.copy(deep=copy)
         try:
             data = self._column.astype(dtype)
+            if copy:
+                data = data.copy(deep=True)
 
-            return self._copy_construct(
-                data=data.copy(deep=True) if copy else data, index=self.index
+            return self._from_data(
+                {self.name: data.copy(deep=True)}, index=self._index
```

Comment:
Is this still ok to do if `copy` is False?

Reply:
Great catch! I had moved the copy logic into a separate line there and forgotten to re-inline it properly, thanks.